projects/ADuCM3029_demo_cn0428_cn0429/src/cli.c

```diff
@@ -0,0 +1,986 @@
+/**
+******************************************************************************
+*   @file     cli.c
+*   @brief    Command Line Interface source file
+*   @version  V0.1
+*   @author   ADI
+*
+*******************************************************************************
+*
+*Copyright 2015-2018(c) Analog Devices, Inc.
+*
+*All rights reserved.
+*
+*Redistribution and use in source and binary forms, with or without modification,
+*are permitted provided that the following conditions are met:
+*    - Redistributions of source code must retain the above copyright
+*      notice, this list of conditions and the following disclaimer.
+*    - Redistributions in binary form must reproduce the above copyright
+*      notice, this list of conditions and the following disclaimer in
+*      the documentation and/or other materials provided with the
+*      distribution.
+*    - Neither the name of Analog Devices, Inc. nor the names of its
+*      contributors may be used to endorse or promote products derived
+*      from this software without specific prior written permission.
+*    - The use of this software may or may not infringe the patent rights
+*      of one or more patent holders.  This license does not release you
+*      from the requirement that you obtain separate licenses from these
+*      patent holders to use this software.
+*    - Use of the software either in source or binary form, must be run
+*      on or directly connected to an Analog Devices Inc. component.
+*
+*THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+*INCLUDING, BUT NOT LIMITED TO, NON-INFRINGEMENT, MERCHANTABILITY AND FITNESS FOR A
+*PARTICULAR PURPOSE ARE DISCLAIMED.
+*
+*IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+*EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, INTELLECTUAL PROPERTY
+*RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR
+*BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+*STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF
+*THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************
+**/
+
+/***************************** Include Files **********************************/
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+#include <ctype.h>
+#include "cli.h"
+#include "ADuCM3029_demo_cn0428_cn0429.h"
+#include "Ringbuffer.h"
+
+/********************************* Definitions ********************************/
+
+
+
+/************************** Variable Definitions ******************************/
+/* Available commands */
+char *CmdCommands[] = {
```

Comment:
regarding the whole `cli.c` functionality, i'm a bit confused/undecided/uncertain/mix-of-some-feelings;
i didn't see this in any projects, so i'm somewhat uncertain about having this here or not;

we could have this [here] only for this project, but then we'd probably get requests to add something similar in other projects ;
depending on the effort to get this done for other projects [and [who]/[if there will be someone] assigned to work on them], we could say yay or nay to this [for now]

personally, i'd be in favor of `yay`, so my following comments will be addressed from that perspective

but even the `nay` [from my perspective] would be more like `nay yet`, since a common code/framework for CLI should be considered